### PR TITLE
LOB-196-event-notifications-flows-v4

### PR DESCRIPTION
**Pull request by alinolteanufr**
- adding converters for event notification flow v4.

### CODE REVIEW

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2020-2024 ForgeRock AS (obst@forgerock.com)
```
Comment (wmorrison365fr) [inline secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:2]: Just 2024 if these are both new?
(Also in other class)

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                .maxEvents(obEventPolling.getMaxEvents())
+                .returnImmediately(obEventPolling.getReturnImmediately())
+                .ack(obEventPolling.getAck())
+                .setErrs(toFREventPollingErrors(obEventPolling.getSetErrs()))
+                .build();
+    }
+
+    public static Map<String, FREventPollingError> toFREventPollingErrors(Map<String, OBEventPolling1SetErrsValue> setErrs) {
+        return setErrs == null ? null : setErrs.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> toFREventPollingError(e.getValue())));
```
Comment (wmorrison365fr) [inline secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:39]: No big deal but you could import static `toMap()`.

We tend to do that in IG core, but maybe you guys don't.

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventSubscriptionConverter.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import uk.org.openbanking.datamodel.v4.event.OBEventSubscription1;
+
+public class FREventSubscriptionConverter {
+
+    // OB to FR
+    public static FREventSubscriptionData toFREventSubscriptionData(OBEventSubscription1 obEventSubscription) {
+        return obEventSubscription == null ? null : FREventSubscriptionData.builder()
+                .callbackUrl(obEventSubscription.getData().getCallbackUrl() != null ? obEventSubscription.getData().getCallbackUrl().toString() : null)
+                .version(obEventSubscription.getData().getVersion())
+                .eventTypes(obEventSubscription.getData().getEventTypes())
```
Comment (wmorrison365fr) [inline secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventSubscriptionConverter.java:28]: Does it matter that it's grabbing the same list from the `obEventSubscription` here? i.e. that the list of `eventTypes` is mutable and could be manipulated on the `ob` side and we'd see the changes in the returned `fr` version?

**Review by wmorrison365fr — APPROVED** [review]
A few comments to check over, but LGTM, thanks.

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java (hunk 2 above)
Comment (wmorrison365fr) [reply secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:39]: Does it get reverted after a successful PR merge of [PR 230](https://github.com/SecureApiGateway/secure-api-gateway-ob-uk-common/pull/230#pullrequestreview-2429340501)?

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventSubscriptionConverter.java (hunk 3 above)
Comment (alinolteanufr) [reply secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventSubscriptionConverter.java:28]: it shouldn't matter, the ob object is parsed one time during the request and the data is converted to the FR object then stored in the database

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java (hunk 1 above)
Comment (alinolteanufr) [reply secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:2]: This is the mandatory format of copyright header

Comment (wmorrison365fr) [reply secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:2]: 2020 though? For a new file... Should it not just be 2024?

**Review by wmorrison365fr — APPROVED** [review]

File: secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java (hunk 1 above)
Comment (alinolteanufr) [reply secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventPollingConverter.java:2]: this is the template in the pipeline at copyright check stage

**Pull request merged**